### DISCUSSION TITLE
not getting any data

### DISCUSSION THREAD

**Discussion by ghost** [ArminJo/JK-BMSToPylontechCAN#21] [Q&A]
hello 
i just want to display the batteries information using just gps port (tx rx) without connecting it to inverter, the data im getting are always like the standalone test without any change. can you help me with it asap
thank you in advance

**Comment by ArminJo** [comment]
No problem. What have you tested so far?

**Comment by ghost** [comment]
I've just connected the gps pins(tx rx Gnd)as per your circuit and I want to display the data on the screen as shown in this picture(i have only 5 batteries)
![WhatsApp Image 2023-11-15 at 11 42 57 AM](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/151038649/da65fe90-939d-4313-a1ad-400d9ec1e9e4)

i didn't modify the code i just upload it on Arduino Uno, I'm getting the standalone values on the screen not the real time data from the bms
I dont want to connect it to an inverter i just want to display all the data related to the batteries and the bms.
my connections:
![WhatsApp Image 2023-11-17 at 9 19 49 AM](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/151038649/34eacace-149d-466b-a8ab-afe70d67e911)
![WhatsApp Image 2023-11-17 at 9 19 49 AM (1)](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/151038649/fb6811c3-a13b-4712-bd06-60364aa381c9)
![WhatsApp Image 2023-11-17 at 9 19 49 AM (2)](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/151038649/bce62a01-ed4e-4661-a067-ed0876b1a852)
please can you help me if the code must be modified or something is wrong with the connection. I want the real time data(voltage, temperature, current ...)
thank you in advance

**Reply by ArminJo** [reply]
What is the Serial output of the program?
is line 155 commented out `//#define STANDALONE_TEST`

**Comment by ghost** [comment]
Version 2.4.0 from Nov 17 2023
Monitoring enabled
Serial to JK-BMS started with 115200 bit/s!
First access to MCP2515 config mode register failed
Starting CAN failed!
Page switching button is at pin 2
At long press, CAN Info page is entered and additional debug data is printed as long as button is pressed
1000 ms between 2 BMS requests
1000 ms between 2 CAN transmissions
LCD Backlight timeout is 30 s

Standalone test. Use fixed demo data

0x00  0x4E 0x57 0x01 0x2D 0x00 0x00 0x00 0x00 0x06 0x00 0x01 
0x0B  0x79 0x30 
0x0D  0x01 0x0C 0xC6 
0x10  0x02 0x0C 0xBE 0x03 0x0C 0xC7 0x04 0x0C 0xC7 0x05 0x0C 0xC7 0x06 0x0C 0xC5 0x07 
0x20  0x0C 0xC6 0x08 0x0C 0xC7 0x09 0x0C 0xC2 0x0A 0x0C 0xC2 0x0B 0x0C 0xC2 0x0C 0x0C 
0x30  0xC2 0x0D 0x0C 0xC1 0x0E 0x0C 0xBE 0x0F 0x0C 0xC1 0x10 0x0C 0xC1 
0x3D  0x80 0x00 0x16 
0x40  0x81 0x00 0x15 0x82 0x00 0x15 0x83 0x14 0x6C 0x84 0x08 0xD0 0x85 0x47 0x86 0x02 
0x50  0x87 0x00 0x04 0x89 0x00 0x00 0x01 0xE0 0x8A 0x00 0x0E 0x8B 0x00 0x00 0x8C 0x00 
0x60  0x07 0x8E 0x16 0x26 0x8F 0x10 0xAE 0x90 0x0F 0xD2 0x91 0x0F 0xA0 0x92 0x00 0x05 
0x70  0x93 0x0B 0xEA 0x94 0x0C 0x1C 0x95 0x00 0x05 0x96 0x01 0x2C 0x97 0x00 0x07 0x98 
0x80  0x00 0x03 0x99 0x00 0x05 0x9A 0x00 0x05 0x9B 0x0C 0xE4 0x9C 0x00 0x08 0x9D 0x01 
0x90  0x9E 0x00 0x5A 0x9F 0x00 0x46 0xA0 0x00 0x64 0xA1 0x00 0x64 0xA2 0x00 0x14 0xA3 
0xA0  0x00 0x46 0xA4 0x00 0x46 0xA5 0xFF 0xEC 0xA6 0xFF 0xF6 0xA7 0xFF 0xEC 0xA8 0xFF 
0xB0  0xF6 0xA9 0x0E 0xAA 0x00 0x00 0x01 0x40 0xAB 0x01 0xAC 0x01 0xAD 0x04 0x11 0xAE 
0xC0  0x01 0xAF 0x01 0xB0 0x00 0x0A 0xB1 0x14 0xB2 0x31 0x32 0x33 0x34 0x35 0x36 0x00 
0xD0  0x00 0x00 0x00 0xB3 0x00 0xB4 0x49 0x6E 0x70 0x75 0x74 0x20 0x55 0x73 0xB5 0x32 
0xE0  0x31 0x30 0x31 0xB6 0x00 0x00 0xE2 0x00 0xB7 0x31 0x31 0x2E 0x58 0x57 0x5F 0x53 
0xF0  0x31 0x31 0x2E 0x32 0x36 0x5F 0x5F 0x5F 0xB8 0x00 0xB9 0x00 0x00 0x04 0x00 0xBA 
0x100  0x49 0x6E 0x70 0x75 0x74 0x20 0x55 0x73 0x65 0x72 0x64 0x61 0x4A 0x4B 0x5F 0x42 
0x110  0x32 0x41 0x32 0x30 0x53 0x32 0x30 0x50 0xC0 0x01 
0x11A  0x00 0x00 0x00 0x00 0x68 0x00 0x00 0x51 0xC2 

*** BMS INFO ***
Protocol Version Number=1
Software Version Number=11.XW_S11.26___
Modify Parameter Password=123456
# External Temperature Sensors=2
*** BATTERY INFO ***
Manufacturer Date=2101
Manufacturer Id=Input UserdaJK_B2A20S20P
Device ID String=Input Us
Device Address=1
Total Battery Capacity[Ah]=320, Low Capacity Alarm Percent=20
Charging Cycles=4
Total Charging Cycle Capacity=480
# Battery Cells=14, Cell Count=14

*** VOLTAGE PROTECTION INFO ***
Battery Overvoltage Protection[mV]=56700, Undervoltage=42700
Cell Overvoltage Protection[mV]=4050, Recovery=4000, Delay[s]=5
Cell Undervoltage Protection[mV]=3050, Recovery=3100, Delay[s]=5
Cell Voltage Difference Protection[mV]=300
Discharging Overcurrent Protection[A]=7, Delay[s]=3
Charging Overcurrent Protection[A]=5, Delay[s]=5

*** TEMPERATURE PROTECTION INFO ***
Power MosFet Temperature Protection=90, Recovery=70
Sensor1 Temperature Protection=100, Recovery=100
Sensor1 to Sensor2 Temperature Difference Protection=20
Charge Overtemperature Protection=70, Discharge=70
Charge Undertemperature Protection=-20, Recovery=-10
Discharge Undertemperature Protection=-20, Recovery=-10

*** MISC INFO ***
Balance Starting Cell Voltage=[mV]3300
Balance Triggering Voltage Difference[mV]=8

Current Calibration[mA]=1041
Sleep Wait Time[s]=10

Dedicated Charge Switch Active=0
Start Current Calibration State=0
Battery Actual Capacity[Ah]=1024

CSV: 270;262;271;271;271;269;270;271;266;266;266;266;265;262;265;265;5228;-22.56;71;1
Total Runtime Minutes=57856 ->   40D04H16M
*** CELL INFO ***
16 Cells, Minimum=3262 mV, Maximum=3271mV, Delta=9 mV, Average=3267 mV
 1=3270 mV,  2=3262 mV,  3=3271 mV,  4=3271 mV,  5=3271 mV,  6=3269 mV,  7=3270 mV,  8=3271 mV, 
 9=3266 mV,  10=3266 mV, 11=3266 mV, 12=3266 mV, 13=3265 mV, 14=3262 mV, 15=3265 mV, 16=3265 mV, 

Warning: CellOvervoltageProtectionMillivolt value 4050 mV > 3450 mV is not recommended for LiFePO4 chemistry. There is less than 1% extra capacity above 3.5V.
Temperature: Power MosFet=22, Sensor 1=21, Sensor 2=21
SOC[%]=71 -> Remaining Capacity[Ah]=227
Battery Voltage[V]=52.28, Current[A]=-22.56, Power[W]=-1179, Difference to full[V]=4.4
Charging MosFet enabled, active | Discharging MosFet enabled, active | Balancing enabled, active
Switch off LCD display, triggered by LCD "ON" timeout reached.

**Comment by ghost** [comment]
![WhatsApp Image 2023-11-17 at 12 14 20 PM](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/151038649/082fa43f-acde-4e6e-ba0e-effa5616a46b)
this picture is live data from bms application connected to the phone using bluetooth i want the same data on lcd or serial
in addition i have 5 ncm (53AH 3.7v each) batteries

**Reply by ArminJo** [reply]
And I want you to check, if line 155 is commented out //#define STANDALONE_TEST.

**Comment by ghost** [comment]
yes it is commented

**Comment by ghost** [comment]
i extracted the file and uploaded the code again it gave me this on the serial 
Version 2.4.0 from Nov 17 2023
Monitoring enabled
Serial to JK-BMS started with 115200 bit/s!
First access to MCP2515 config mode register failed
Starting CAN failed!
Page switching button is at pin 2
At long press, CAN Info page is entered and additional debug data is printed as long as button is pressed
2000 ms between 2 BMS requests
2000 ms between 2 CAN transmissions
LCD Backlight timeout is 5 min

Receive timeout at ReplyFrameBufferIndex=0

**Reply by ArminJo** [reply]
Please do not waste my time!
Test carefully before asking for support!

**Comment by ghost** [comment]


**Reply by ArminJo** [reply]
How old are you?

**Comment by ghost** [comment]


**Reply by ArminJo** [reply]
Now I understand...
